### DISCUSSION TITLE
compiled/installed nginx 1.26.1 claims to be 1.18 in http query?

### DISCUSSION THREAD

**Discussion by goofyseeker311** [nginx/nginx#310] [General]
somehow a http query to the nginx 1.26.1 bind port results in:
```
HTTP/1.1 200 OK
Server: nginx/1.18.0
...
```

I did install and remove nginx 1.18 from apt store on debian before compiling/installing 1.26.1 from source.
nginx -v also says 1.18 even if its no longer installed. so where nginx gets its version value if not from the binary?

edit: I have the arut/nginx-rtmp-module configured as additional module.

**Comment by viktorberezikov** [comment]
The nginx version is compiled/stored in a binary file. [For example](https://github.com/nginx/nginx/blob/29aec5720fdfc74dca8d99d5cf6dc0fcb4e4ce2f/src/core/nginx.h#L12). Maybe you have multiple binaries. You can find the binary file you are accessing by running the command: `which nginx`.

**Comment by goofyseeker311** [comment]
no, even the compiled binary in the objs directory says 1.18 and not 1.26.1. which points to the make install version of nginx. so the compile takes version number from somewhere else than the code, which says also 1.26.1.

edit: the compiled version of nginx run by ./nginx says 1.26.1, so I only conclude that "sudo make install" script fails to actually install nginx as the primary nginx.

**Comment by Maryna-f5** [comment]
This discussion has been inactive for more than 90 days, closing it.
